src/selectors/entities/channel_categories.ts

```diff
@@ -0,0 +1,250 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {createSelector} from 'reselect';
+
+import {General, Preferences} from '../../constants';
+import {CategoryTypes} from '../../constants/channel_categories';
+
+import {getCurrentChannelId} from 'selectors/entities/channels';
+import {getCurrentUserLocale} from 'selectors/entities/i18n';
+import {getTeammateNameDisplaySetting, shouldAutocloseDMs} from 'selectors/entities/preferences';
+import {getCurrentUserId} from 'selectors/entities/users';
+
+import {Channel} from 'types/channels';
+import {ChannelCategory} from 'types/channel_categories';
+import {GlobalState} from 'types/store';
+import {UserProfile} from 'types/users';
+import {IDMappedObjects, RelationOneToOne} from 'types/utilities';
+
+import {getUserIdFromChannelName, isFavoriteChannel} from 'utils/channel_utils';
+import {getPreferenceKey} from 'utils/preference_utils';
+import {displayUsername} from 'utils/user_utils';
+
+export function getCategoryIdsForTeam(state: GlobalState, teamId: string): string[] | undefined {
+    return state.entities.channelCategories.orderByTeam[teamId];
+}
+
+export function makeGetCategoriesForTeam() {
+    return createSelector(
+        getCategoryIdsForTeam,
+        (state: GlobalState) => state.entities.channelCategories.byId,
+        (categoryIds, categoriesById) => {
+            if (!categoryIds) {
+                return [];
+            }
+
+            return categoryIds.map((id) => categoriesById[id]);
+        }
+    );
+}
+
+export function makeGetChannelsForAllCategories() {
+    return createSelector(
+        (state: GlobalState) => state.entities.channels.channels,
+        (state: GlobalState, teamId: string) => teamId,
+        (allChannels: IDMappedObjects<Channel>, teamId: string) => {
+            return Object.values(allChannels).filter((channel) => channel.team_id === teamId || channel.team_id === '');
+        }
+    );
+}
+
+export function makeFilterChannelsByFavorites() {
+    return createSelector(
+        (state: GlobalState, channels: Channel[]) => channels,
+        (state: GlobalState, channels: Channel[], categoryType: string) => categoryType,
+        (state: GlobalState) => state.entities.preferences.myPreferences,
+        (channels, categoryType, myPreferences) => {
+            const filtered = channels.filter((channel) => {
+                if (categoryType === CategoryTypes.FAVORITES) {
```

Comment:
Could this just be simplified to:
```suggestion
                return categoryType === CategoryTypes.FAVORITES && isFavoriteChannel(myPreferences, channel.id);
```

Reply:
No, that wouldn't work since it would make it always return false when the category isn't Favourites. `(categoryType === CategoryTypes.FAVORITES) === isFavoriteChannel(myPreferences, channel.id)` is more like what you're looking for, but that just looks awful with the nested equals.